llvm/lib/Target/AArch64/Utils/AArch64BaseInfo.h

```diff
@@ -700,8 +700,8 @@ AArch64StringToVectorLayout(StringRef LayoutStr) {
 
 namespace AArch64SysReg {
   struct SysReg {
-    const char *Name;
-    const char *AltName;
+    const char Name[32];
```

Comment:
What about using StringLiteral?

Reply:
Under the hood, a `StringLiteral` is still a pointer and a size, and the pointer has to be dynamically relocated so it won't achieve the goal of the PR.